### DISCUSSION TITLE
Help Wanted: dbatools 2.0

### DISCUSSION THREAD

**Discussion by potatoqualitee** [dataplat/dbatools#8696] [Contributors]
I'm totally burned out and having a tough time with getting 2.0 out the door. I know its the holiday season, so it's terrible timing but if anyone has time to help with some of those issues, I'd appreciate it

[These are the issues](https://github.com/dataplat/dbatools/issues?q=is%3Aissue+is%3Aopen+label%3A%22fix+in+2.0%22) that are left for dbatools 2.0.

I'll also need a blitz of blog posts so if anyone can write about

* The changes that Microsoft has made to SqlClient which sets `Encrypt=True` and `TrustServerCertificate=false` by default. This will cause people to have immediate issues with dbatools 2.0 so I want to bring awareness and set expectations.
  * Azure Data Studio has the same issue (but not SSMS for now) and the team has written about it here: https://learn.microsoft.com/en-us/sql/azure-data-studio/connect
  * This can be addressed in dbatools with the following commands, (but people should work to secure their machines instead and I needa blog post about that too bc dbatools makes it super easy to manage certs and force network encryption):
  * `Set-DbatoolsConfig -FullName sql.connection.trustcert -Value $true -Register`
  * `Set-DbatoolsConfig -FullName sql.connection.encrypt -Value $false -Register`
* <strike>I'll also need blog posts about the new structure of dbatools. Most people will be able to just `Install-Module dbatools` like always but people using PowerShell v3 or v4 will need to `Install-Module dbatools.legacy` which is _EXACTLY_ the same _EXCEPT_ the Module Manifest (dbatools.psd1).</strike>
  * This is because we've separated out the dbatools and SMO libraries. This makes dbatools smaller overall because 
    * People will just have to install the DLLs once until the library itself is updated
  * This does impact the way people install offline. They can't just download the git repo as the dbatools.library module is built & compiled in GitHub Actions but the binaries are not stored in the repo -- they are stored in the PowerShell Gallery. This means that people will have to have access to `Save-Module` or do some manual downloading from the PS Gallery and rename the .nuget to .zip, extract, and rename the directories as appropriate.

If I can't get this out before the end of the year, I leave for France at the beginning of January and 2.0 wont be released till much later.

For dbatools developers, 2.0 is immediately different, as you'll need to Install-Module dbatools.library before attempting to develop.

So if you can close some 2.0 issues in the repo or draft some blog posts in preparation for 2.0, I'd appreciate it :pray:

**Comment by potatoqualitee** [comment]
btw I need a blitz of blog posts so the more people writing the better, sorta like what we did for dbachecks. if you find out someone else is writing about it, please dont let that stop you!

Also, for those running on PowerShell Core, you must be running version 7.2 or above. I need to make a table and once I do, anyone is free to nab.

**Comment by ACALVETT** [comment]
Just seen this.  

I will put a blog together around certificates as you requested. I can pop the blog up on my site although its been dormant for a long time or i can supply the content to go onto dbaTools. Will ping once its done.

I see https://github.com/dataplat/dbatools/issues/8697. 

So will comment there and put something together

**Reply by potatoqualitee** [reply]
That's awesome! Thank you so much @ACALVETT 💞

**Reply by potatoqualitee** [reply]
Things have changed, I should update this issue. Because of the issue hilighted in @wsmelton's Discussion, the library stuff no longer applies (glad then you are working on certs, that is still valid!)

https://github.com/dataplat/dbatools/discussions/8715

I'm writing a spicy blog post now about what went wrong with the library module implementation.

**Reply by potatoqualitee** [reply]
I've written a bunch of posts at https://blog.netnerds.net if you're interested in checking them out, @ACALVETT
